packages/rich-text-editor/src/RichTextEditor/RichTextEditor.tsx

```diff
@@ -56,25 +58,41 @@ export const RichTextEditor: React.VFC<RichTextEditorProps> = props => {
     classNameOverride,
     controls,
     rows = 3,
+    dataError = <p>Something went wrong</p>,
+    onDataError,
     ...restProps
   } = props
   const [schema] = useState<Schema>(createSchemaFromControls(controls))
   const [labelId] = useState<string>(labelledBy || v4())
   const [editorId] = useState<string>(v4())
 
-  const [editorRef, editorState, dispatchTransaction] = useRichTextEditor(
-    EditorState.create({
-      doc: value
-        ? Node.fromJSON(schema, {
-            type: "doc",
-            content: value,
-          })
-        : null,
-      schema,
-      plugins: getPlugins(controls, schema),
-    }),
-    { "aria-labelledby": labelId, role: "textbox" }
-  )
+  const useRichTextEditorResult = (() => {
+    try {
+      return useRichTextEditor(
+        EditorState.create({
+          doc: value
+            ? Node.fromJSON(schema, {
+                type: "doc",
+                content: value,
+              })
+            : null,
+          schema,
+          plugins: getPlugins(controls, schema),
+        }),
+        { "aria-labelledby": labelId, role: "textbox" }
+      )
+    } catch {
+      return new Error("Bad data error")
+    }
+  })()
+
+  if (useRichTextEditorResult instanceof Error) {
+    onDataError && onDataError()
+    return dataError
```

Comment:
We could render the inline notification here and pass in the dataError in as a child